src/atomic_ptr.hpp

```diff
@@ -244,6 +244,12 @@ struct atomic_value_t
 
     atomic_value_t (const atomic_value_t &src) : value (src.load ()) {}
 
+    atomic_value_t &operator= (const atomic_value_t &src)
```

Comment:
I think atomic_value_t was non-copyable by intention. Is this really necessary? If it is, probably the code location causing this should be fixed.

Reply:
I think the WindRiver Diab compiler needed me to declare and define the `operator=` because something needed it. I can try to see why it needed it.